internal/pipeline/executor.go

```diff
@@ -283,18 +288,22 @@ func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
 
 // fnFilters returns chain of functions that are applicable
 // to a given pipeline.
-func fnFilters(_ *Pipeline) []kio.Filter {
-	fn := &annotator{
-		key:   "builtin/setter-1",
-		value: "test",
+func fnFilters(p *Pipeline) ([]kio.Filter, error) {
+	filters, err := p.fnChain()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get function chain: %w", err)
 	}
 
-	// TODO(droot): Implement this with the logic to create
-	// function chain from a pipeline
-	// hardcoding built-in set-annotation function for testing
-	return []kio.Filter{
-		&fnRunner{
-			fn: fn,
-		},
+	var ans []kio.Filter
+	for _, f := range filters {
+		ans = append(ans, f)
 	}
+
+	ans = append(ans, &fnRunner{
```

Comment:
I am thinking may be we can support some builtin functions with special image path `set-annotation`, that can help us test the pipeline (without hardcoding) without running docker.

Reply:
How about `kpt.dev/builtin/set-annotation`?